plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -249,7 +249,7 @@ void deleteBlobs(Collection<String> blobNames) throws IOException {
             deleteBlob(blobNames.iterator().next());
             return;
         }
-        final List<Storage.Objects.Delete> deletions = new ArrayList<>();
+        final List<Storage.Objects.Delete> deletions = new ArrayList<>(blobNames.size());
```

Comment:
@nik9000 Looks like if the list reaches MAX_BATCHING_REQUESTS it will get cleared before doing another iteration of the loop. So I've updated it to that.